src/main/resources/script.sh

```diff
@@ -0,0 +1,1353 @@
+        if [ $? != 0 ]; then
+            exist="false"
+        fi
+        i=$(expr $i + 1)
+    done
+    info "Removing all cache entries for ${LFN} (files will stay locally in case anyone else needs them)"
+    local TEMP=$(mktemp temp.XXXXXX)
+    awk -v L="${LFN}" '$1!=L {print}' "$cacheDir/$cacheFile" > "${TEMP}"
+    \mv -f "${TEMP}" "$cacheDir/$cacheFile"
+    info "Adding file ${FILE} to cache and setting the timestamp"
+    \cp -f "${FILE}" "${NAME}"
+    local date_local=$(ls -la "${NAME}" | awk -F' ' '{print $6, $7, $8}')
+    local TIMESTAMP=$(date -d "${date_local}" +%s)
+    echo "${LFN} ${NAME} ${TIMESTAMP}" >> "$cacheDir/$cacheFile"
+}
+
+export -f addToCache
+
+
+function addToFailOver {
+    local LFN="$1"
+    local FILE="$2"
+    local REMOTEFILE=$(lcg-lr "lfn:${LFN}" | grep "$failOverHost")
+    generated=${REMOTEFILE#*generated}
+    local RPFILE="${generated}"
+
+    lcg-del --nobdii --defaultsetype srmv2 -v "srm://$failOverHost:$failOverPort/srm/managerv2?SFN=$failOverHome${RPFILE}" &>/dev/null
+    lfc-ls "${LFN}"
+    if [ $? = 0 ]; then
+        lfc-rename "${LFN}" "${LFN}-garbage-$(date +"%Y-%m-%d-%H-%M-%S")"
+    fi
+    lfc-mkdir -p "$(dirname "${LFN}")"
+    local FILENAME=$(echo "$RANDOM$RANDOM" | md5sum | awk '{print $1}')
+    local FOLDERNAME=$(date +"%Y-%m-%d")
+    local OPTS="--nobdii --defaultsetype srmv2"
+    DM_DEST="srm://$failOverHost:$failOverPort/srm/managerv2?SFN=$failOverHome/${FOLDERNAME}/file-${FILENAME}"
+    GUID=$(lcg-cr ${OPTS} -d "${DM_DEST}" "file:${FILE}")
+    if [ $? = 0 ]; then
+        lcg-aa "${GUID}" "lfn:${LFN}"
+        if [ $? = 0 ]; then
+            info "Data successfully copied to Fail Over."
+        else
+            error "Unable to create LFN alias ${LFN} to ${GUID}"
+            return 1
+        fi
+    else
+        error "Unable to copy data to Fail Over."
+        return 1
+    fi
+}
+
+export -f addToFailOver
+
+
+#!/bin/bash
+
+nSEs() {
+    local i=0
+    for n in ${SELIST}; do
+        i=$((i + 1))
+    done
+    return $i
+}
+
+getAndRemoveSE() {
+    local index=$1
+    local i=0
+    local NSE=""
+    RESULT=""
+    for n in ${SELIST}; do
+        if [ "$i" = "${index}" ]; then
+            RESULT=$n
+            info "result: $RESULT"
+        else
+            NSE="${NSE} $n"
+        fi
+        i=$((i + 1))
+    done
+    SELIST=${NSE}
+    return 0
+}
+
+chooseRandomSE() {
+    nSEs
+    local n=$?
+    if [ "$n" = "0" ]; then
+        info "SE list is empty"
+        RESULT=""
+    else
+        local r=${RANDOM}
+        local id=$((r % n))
+        getAndRemoveSE ${id}
+    fi
+}
+
+uploadLfnFile() {
+    local LFN=$1
+    local FILE=$2
+    local nrep=$3
+    local SELIST=${SE}
+
+    
+
+    LFN=$(echo ${LFN} | sed -r -e 's/^lfn://' -e 's#//#/#g')
+
+    info "getting file size and computing sendReceiveTimeout"
+    local size=$(ls -l ${FILE} | awk -F' ' '{print $5}')
+    local sendReceiveTimeout=$((size / minAvgDownloadThroughput / 1024))
```

Comment:
in develop it is `echo $[${size:-0}/${minAvgDownloadThroughput}/1024]`
Have you changed that to improve ?

Reply:
In all case, bring back the minAvgDownloadThroughput variable